### PR TITLE
Document potential data loss when changing metric types

### PR DESCRIPTION
**Pull request by Dexterp37**
This additionally adds test coverage as requested by @badboy

### CODE REVIEW

**Comment by badboy** [comment]
gnah, the schema check failed:
```
curl: (22) The requested URL returned error: 403 rate limit exceeded
```

This was not something I anticipated.

**Comment by badboy** [comment]
(it's not a required check, so you can ignore it here)

**Comment by Dexterp37** [comment]
> 
> 
> gnah, the schema check failed:
> 
> ```
> curl: (22) The requested URL returned error: 403 rate limit exceeded
> ```
> 
> This was not something I anticipated.

Interesting, we're not even pushing that many PRs at this moment in time.

**Comment by badboy** [comment]
> Interesting, we're not even pushing that many PRs at this moment in time.

It's coming from CircleCI servers toward GitHub, it's not just us (we _might_ be able to pass along our api token to go around the limits, but ... hm)

File: docs/user/metric-parameters.md [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 - `type`: **Required.**  Specifies the type of a metric, like "counter" or "event". This defines which operations are valid for the metric, how it is stored and how data analysis tooling displays it. See the list of [supported metric types](metrics/index.md).
 
+> **Important**: Once a metric is released in a product, its `type` should not be changed. If any data was collected locally with the older `type`, before the ping it goes in has a chance to be submitted, recording data with the new `type` may cause any old persisted data to be lost for that metric. See [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1621757#c1) for an extended explanation of the different scenarios.
```
Comment (mdboom) [inline docs/user/metric-parameters.md]:
```suggestion
> **Important**: Once a metric is released in a product, its `type` should not be changed. If any data was collected locally with the older `type`, and hasn't yet been sent in a ping, recording data with the new `type` may cause any old persisted data to be lost for that metric. See [this comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1621757#c1) for an extended explanation of the different scenarios.
```

Comment (mdboom) [reply docs/user/metric-parameters.md]: Slightly clearer to me, maybe, but not incorrect either way.

**Pull request merged**